benchmarks/scripts/expval_benchmark.py

```diff
@@ -1,99 +1,148 @@
+    error = depolarizing_error(0.01, 1)
+    two_qubit_error = depolarizing_error(0.02, 2)
+    # TODO: errors should only be added to the gateset that we are compiling to
+    # but there is a bug with cirq currently compiling to U3 and CZ
+    depolarizing_noise.add_all_qubit_quantum_error(
+        error, ["u1", "u2", "u3", "rx", "ry", "rz", "h"]
+    )
+    depolarizing_noise.add_all_qubit_quantum_error(
+        two_qubit_error, ["cx", "cz"]
+    )
 
     simulator = AerSimulator(
         method="density_matrix", noise_model=depolarizing_noise
     )
     return simulator.run(circuit).result().data()["density_matrix"]
 
 
-density_matrices = {
-    compiler: simulate_density_matrix(circuit)
-    for compiler, circuit in generate_compiled_circuits(qasm_string).items()
-}
+def qiskit_gateset(circuit: qiskit.QuantumCircuit) -> set[str]:
+    everything = set(circuit.count_ops().keys())
+    return everything - {"save_density_matrix"}
+
+
+uncompiled_qiskit_circuit = get_native_rep(qasm_string, "qiskit")
+native_circuit = get_native_rep(qasm_string, compiler_alias)
+compiled_circuit = compile_for_simulation(native_circuit, compiler_alias)
+
+if log:
+    circuit_name = os.path.split(qasm_path)[-1]
+    print(f"Compiling {circuit_name} with {compiler_alias}")
+    print(
+        f"    Gate reduction: {len(uncompiled_qiskit_circuit)} -> {len(compiled_circuit) - 1}"
+    )  # minus 1 to account for the addition of `save_density_matrix`
+    print(
+        f"    Starting gate set: {qiskit_gateset(uncompiled_qiskit_circuit)}"
+    )
+    print(f"    Final gate set:    {qiskit_gateset(compiled_circuit)}")
+    print(f"    Starting gates: {uncompiled_qiskit_circuit.count_ops()}")
+    print(f"    Final gates:    {compiled_circuit.count_ops()}")
 
+density_matrix = simulate_density_matrix(compiled_circuit)
 
-observable = Operator.from_label("ZZZZZZZZZZ")
+obs_str = "Z" * compiled_circuit.num_qubits
+observable = Operator.from_label(obs_str)
 
-expectation_values = {
-    compiler: np.real(dm.expectation_value(observable))
-    for compiler, dm in density_matrices.items()
-}
+compiled_ev = np.real(density_matrix.expectation_value(observable))
 
 
-ideal_circuit = qasm2.loads(qasm_string)
-ideal_state = Statevector.from_instruction(ideal_circuit)
-ideal_expval = np.real(ideal_state.expectation_value(observable))
+ideal_state = Statevector.from_instruction(uncompiled_qiskit_circuit)
+ideal_ev = np.real(ideal_state.expectation_value(observable))
 
 results = [
     {
-        "compiler": compiler,
-        "expval": expval,
-        "absoluate_error": abs(ideal_expval - expval),
-        "relative_error": abs(ideal_expval - expval) / abs(ideal_expval),
-        "ideal": ideal_expval,
+        "compiler": compiler_alias,
+        "circuit_name": qasm_path.split("/")[-1].split("_N")[0],
+        "observable": obs_str,
+        "expval": compiled_ev,
+        "absoluate_error": abs(ideal_ev - compiled_ev),
```

Comment:
Nit -- typo here should be `absolute_error` versus `absoluate_error`

Reply:
Good catch!